### PR TITLE
Docker image size fixes

### PR DESCRIPTION
**Pull request by Kagee**
Image talks for itself?
2023-06-02-151809_665x185_scrot

(BCOMMIT represents what build repo commit was used when building the image)

### CODE REVIEW

File: server/Dockerfile [hunk 1]
```diff
@@ -155,9 +156,9 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim"
     COPY --from=php-build /usr/lib/php/${PHP_VER}/brotli.so /usr/lib/php/${PHP_VER}/brotli.so
     COPY --from=php-build /usr/lib/php/${PHP_VER}/simdjson.so /usr/lib/php/${PHP_VER}/simdjson.so
 
-    COPY --from=composer-build /tmp/Vendor /var/www/MISP/app/Vendor
-    COPY --from=composer-build /tmp/Plugin /var/www/MISP/app/Plugin
-    
+    COPY --from=composer-build --chown=www-data:www-data /tmp/Vendor /var/www/MISP/app/Vendor
```
Comment (ostefano) [inline server/Dockerfile:159]: Is this necessary since we chown later?

File: server/Dockerfile [hunk 2]
```diff
@@ -169,17 +170,18 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim"
...
+        # Make a copy of the configurations, so we can sync from it
+        cp -R /var/www/MISP/app/Config /var/www/MISP/app/Config.dist;
+
+    # Combine find+chown/mod-RUNs to not explode image size
+        # The spirit of the upstrem dockerization is to keep user and group aligned in terms of permissions
+    RUN find /var/www/MISP \( ! -user www-data -or ! -group www-data \) -exec chown www-data:www-data '{}' +; \
+        # Files are also executable and read only, because we have some rogue scripts like 'cake' and we can not do a full inventory
+        find /var/www/MISP -not -perm 550 -type f -exec chmod 0550 '{}' +; \
+        # Directories are also writable, because there seems to be a requirement to add new files every once in a while
+        find /var/www/MISP -not -perm 770 -type d -exec chmod 0770 '{}' +;
```
Comment (ostefano) [inline server/Dockerfile:184]: All these look reasonable

File: server/Dockerfile [hunk 3]
```diff
@@ -138,12 +138,13 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim"
         && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
 
     # Download MISP using git in the /var/www/ directory.
-    RUN if [ ! -z ${MISP_COMMIT} ]; then \
+    RUN if [ -n ${MISP_COMMIT} ]; then \
```
Comment (ostefano) [inline server/Dockerfile:141]: Fair

**Comment by ostefano** [comment]
LGTM, just one nit @Kagee

File: server/Dockerfile (hunk 1 above)
Comment (Kagee) [reply server/Dockerfile:159]: This means we move *and* chown the files in the same step, i.e. they are not changed *again* in the later step, increasing the size then.

**Comment by ostefano** [comment]
@Kagee looks like submodule checks depend on the `.git` dir. 
Do you know what functionalities would we be losing?

```
[Exception] An error has occurred while attempt to read file `/var/www/MISP/.git/HEAD`: file doesn't exists.
Stack Trace:
#0 /var/www/MISP/app/Lib/Tools/GitTool.php(49): FileAccessTool::readFromFile()
#1 /var/www/MISP/app/Model/AppModel.php(3240): GitTool::currentCommit()
#2 /var/www/MISP/app/Model/Server.php(4149): AppModel->checkMIPSCommit()
#3 /var/www/MISP/app/Controller/ServersController.php(1109): Server->getCurrentGitStatus()
#4 [internal function]: ServersController->serverSettings()
#5 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#8 /var/www/MISP/app/webroot/index.php(99): Dispatcher->dispatch()
#9 {main}
2023-06-03 18:14:11 Notice: Could not fetch git submodules status
[ProcessException] Command 'git submodule status --cached' finished with error code 128.
STDERR: 'fatal: not a git repository (or any of the parent directories): .git
'
STDOUT: ''
Stack Trace:
#0 /var/www/MISP/app/Lib/Tools/GitTool.php(82): ProcessTool::execute()
#1 /var/www/MISP/app/Model/Server.php(4186): GitTool::submoduleStatus()
#2 /var/www/MISP/app/Controller/ServersController.php(1449): Server->getSubmodulesGitStatus()
#3 [internal function]: ServersController->getSubmodulesStatus()
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs()
#5 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#7 /var/www/MISP/app/webroot/index.php(99): Dispatcher->dispatch()
#8 {main}
```

**Comment by Kagee** [comment]
I think my CI build may be doing a full clone of MISP since i use MISP_COMMIT, thus the big change when deleting the .git dir. I'll look into either patching the Dockerfile to do a shallow clone anyway, or another workaround.

**Comment by ostefano** [comment]
@Kagee after talking with `igloska`, it seems this might be an acceptable loss of functionality

**Comment by Kagee** [comment]
One should maybe look at making some of the diag elemements controllable, so they could be disabled when using a Docker build.

**Comment by ostefano** [comment]
That was exactly his point.

**Comment by ostefano** [comment]
I'll merge everything but the .git rm.

We can think of that once we take care of this on the MISP side.

**Comment by ostefano** [comment]
Squashed and merged together with other fixes. Check https://github.com/ostefano/docker-misp/commit/b46e9eef6304b4ed946b9de5e553b20573725705

**Pull request closed without merging**